### PR TITLE
Hw dao3

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

**Review by Artificial-Friend — APPROVED** [review]
Good job!

File: src/main/java/core/Application.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private static final Injector injector =
+            Injector.getInstance(Application.class.getPackageName());
+
+    public static void main(String[] args) {
+        ManufacturerService manufacturerService = (ManufacturerService)
+                injector.getInstance(ManufacturerService.class);
+        Manufacturer mercedes = new Manufacturer("Mercedes", "Germany");
+        Manufacturer volvo = new Manufacturer("Volvo", "Sweden");
+        Manufacturer audi = new Manufacturer("Audi", "Germany");
+        audi.setId(1L);
```
Comment (Sofasmile) [inline src/main/java/core/Application.java:17]: do we need this code `audi.setId(1L)`? method `create` should set `id` to the object

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        Manufacturer updateManufacturer = getAll().stream()
+                .filter(m -> m.getId().equals(manufacturer.getId()))
+                .findFirst()
+                .get();
+        updateManufacturer.setCountry(manufacturer.getCountry());
+        updateManufacturer.setName(manufacturer.getName());
```
Comment (Sofasmile) [inline src/main/java/core/dao/ManufacturerDaoImpl.java]: this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return updateManufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        Manufacturer manufacturer = getAll()
```
Comment (Sofasmile) [inline src/main/java/core/dao/ManufacturerDaoImpl.java]: read about method `removeIf`

File: src/main/java/core/db/Storage.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package core.db;
+
+import core.model.Manufacturer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static List<Manufacturer> manufacturers = new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/core/db/Storage.java]: let's add `final` keyword

File: src/main/java/core/model/Car.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.List;
+import java.util.Objects;
+
+public class Car {
+    private Long id;
+    private String model;
+    private Manufacturer manufacturer;
+    private List<Driver> drivers;
+
+    public Car(String model, Manufacturer manufacturer, List<Driver> drivers) {
```
Comment (Sofasmile) [inline src/main/java/core/model/Car.java]:
```suggestion
    public Car(String model, Manufacturer manufacturer) {
```

File: src/main/java/core/model/Car.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        this.model = model;
+        this.manufacturer = manufacturer;
+        this.drivers = drivers;
```
Comment (Sofasmile) [inline src/main/java/core/model/Car.java]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/03_dao_service_checklist.html
last point from checklist

File: src/main/java/core/service/ManufacturerServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import core.dao.ManufacturerDao;
+import core.lib.Inject;
+import core.lib.Service;
+import core.model.Manufacturer;
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (Sofasmile) [inline src/main/java/core/service/ManufacturerServiceImpl.java]: how about access modifier?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/service/ManufacturerServiceImpl.java (hunk 7 above)
Comment (IwalkInSilence) [reply src/main/java/core/service/ManufacturerServiceImpl.java]: fix

File: src/main/java/core/model/Car.java (hunk 6 above)
Comment (IwalkInSilence) [reply src/main/java/core/model/Car.java]: fix

File: src/main/java/core/model/Car.java (hunk 5 above)
Comment (IwalkInSilence) [reply src/main/java/core/model/Car.java]: fix

File: src/main/java/core/db/Storage.java (hunk 4 above)
Comment (IwalkInSilence) [reply src/main/java/core/db/Storage.java]: done

File: src/main/java/core/Application.java (hunk 1 above)
Comment (IwalkInSilence) [reply src/main/java/core/Application.java:17]: I want to check how my method update working with same id

File: src/main/java/core/dao/ManufacturerDaoImpl.java (hunk 2 above)
Comment (IwalkInSilence) [reply src/main/java/core/dao/ManufacturerDaoImpl.java]: done

File: src/main/java/core/dao/ManufacturerDaoImpl.java (hunk 3 above)
Comment (IwalkInSilence) [reply src/main/java/core/dao/ManufacturerDaoImpl.java]: fix

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -28,22 +28,11 @@ public Manufacturer create(Manufacturer manufacturer) {
...
     @Override
     public Manufacturer update(Manufacturer manufacturer) {
-        Manufacturer updateManufacturer = getAll().stream()
-                .filter(m -> m.getId().equals(manufacturer.getId()))
-                .findFirst()
-                .get();
-        updateManufacturer.setCountry(manufacturer.getCountry());
-        updateManufacturer.setName(manufacturer.getName());
-        return updateManufacturer;
+        return getAll().set(manufacturer.getId().intValue() - 1, manufacturer);
```
Comment (Sofasmile) [inline src/main/java/core/dao/ManufacturerDaoImpl.java]: does this method work correctly?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -28,7 +28,12 @@ public Manufacturer create(Manufacturer manufacturer) {
 
     @Override
     public Manufacturer update(Manufacturer manufacturer) {
-        return getAll().set(manufacturer.getId().intValue() - 1, manufacturer);
+        Manufacturer updateManufacturer = getAll().stream()
+                .filter(m -> m.getId().equals(manufacturer.getId()))
+                .findFirst()
+                .get();
```
Comment (Sofasmile) [inline src/main/java/core/dao/ManufacturerDaoImpl.java:34]:
```suggestion
        Manufacturer updateManufacturer = get(manufacturer.getId()).get();
```

File: src/main/java/core/dao/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -28,7 +28,12 @@ public Manufacturer create(Manufacturer manufacturer) {
...
+        int indexOfUpdatedManufacturer = getAll().indexOf(updateManufacturer);
+        return getAll().set(indexOfUpdatedManufacturer, manufacturer);
```
Comment (Sofasmile) [inline src/main/java/core/dao/ManufacturerDaoImpl.java]: we shouldn't return old value, let's return new

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/dao/ManufacturerDaoImpl.java (hunk 10 above)
Comment (IwalkInSilence) [reply src/main/java/core/dao/ManufacturerDaoImpl.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**